```
FAILED tests/test_spawner.py::test_ipv6_addr - KeyError: 'spec'
```
Most likely because this test uses a mock
https://github.com/minrk/kubespawner/blob/2db388cd82194bef10bd337c9722ffd908673abf/tests/test_spawner.py#L1929-L1931

--- a/tests/test_spawner.py
+++ b/tests/test_spawner.py
@@ -1932,8 +1932,15 @@ async def test_ipv6_addr():
     url = spawner._get_pod_url(
         {
             "metadata": {
-                "annotations": {},
                 "name": "jupyter-test",
+            },
+            "spec": {
+                "containers": [
+                    {
+                        "name": "notebook",
+                        "ports": [{"name": "notebook-port", "containerPort": 8888}],
+                    }
+                ]
             },
             "status": {"podIP": "cafe:f00d::"},
         }

